service_configuration_lib/spark_config.py

```diff
@@ -430,14 +430,16 @@ def _adjust_spark_requested_resources(
         # TODO(gcoll|COREML-2697): Consider cleaning this part of the code up
         # once mesos is not longer around at Yelp.
         if 'spark.executor.instances' not in user_spark_opts:
-
-            executor_instances = int(user_spark_opts.get(
-                'spark.cores.max',
-                str(DEFAULT_MAX_CORES),
-            )) // executor_cores
+            # spark.cores.max provided, calculate based on (max cores // per-executor cores)
+            if 'spark.cores.max' in user_spark_opts:
+                executor_instances = int(user_spark_opts['spark.cores.max']) // executor_cores
```

Comment:
We need to go in the direction, that `spark.cores.max` should not be provided.

Add a warning, that this should be replaced with config, `spark.executor.instances`. 
I think once we have a warning log, maybe in future we could override this config for all jobs.

Reply:
I think the `spark.executor.instances` can still come up to be 0, right?
Example: spark.cores.max=1, spark.executor.cores=2
Maybe in the end set: spark.executor.instaces = max(calculated_one, 1)?

Reply:
Added a warning. Regarding the second comment, if the executor instances come up to be zero when both spark.cores.max and spark.executor.cores are provided, it will throw an error on line 478